core/fetcher.go

```diff
@@ -2,18 +2,27 @@ package core
 
 import (
 	"context"
-	"errors"
 	"fmt"
+	"io"
 
+	"github.com/gogo/protobuf/proto"
 	logging "github.com/ipfs/go-log/v2"
-	coretypes "github.com/tendermint/tendermint/rpc/core/types"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	coregrpc "github.com/tendermint/tendermint/rpc/grpc"
 	"github.com/tendermint/tendermint/types"
 
 	libhead "github.com/celestiaorg/go-header"
 )
 
 const newBlockSubscriber = "NewBlock/Events"
 
+type SignedBlock struct {
```

Comment:
this is the redefinition of the SignedBlock to fix the memory leak we had.
More details in: https://github.com/celestiaorg/celestia-node/pull/3926/files#r1833766092